### PR TITLE
MAGENTO-2060: Added api endpoint for deleting customer addresses

### PR DESCRIPTION
**Pull request by andre-kraus**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I updated the CHANGELOG.md

### CODE REVIEW

File: src/app/code/community/Shopgate/Cloudapi/Model/Api2/Customers/Addresses/Rest/Customer/V2.php [hunk 1]
```diff
@@ -64,4 +69,18 @@ protected function _loadCustomerById($id)
 
         return $customer;
     }
+
+    /**
+     * @param array $data
+     *
+     * @throws Mage_Api2_Exception
+     */
+    protected function _multiDelete(array $data)
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/Model/Api2/Customers/Addresses/Rest/Customer/V2.php]: any reason it is in this class? You see, logic of this class only happens for the customer caller. Hence why there are checks like `$this->getApiUser()->getUserId() !== $customer->getId()` in the other functions. They make sure no malicious user is trying to create/retrieve/delete someone else's stuff. The multi-delete function may also be needed for the admin user call when we have an admin panel or something, so it does not really belong here.

File: src/app/code/community/Shopgate/Cloudapi/etc/api2.xml [hunk 2]
```diff
@@ -258,7 +259,14 @@
                         <route>/v2/customers/:customer_id/addresses</route>
                         <action_type>collection</action_type>
                     </route_v2_customers_id_addresses>
+                    <route_v2_customers_id_addresses_id>
+                        <route>/v2/customers/:customer_id/addresses/:id</route>
+                        <action_type>entity</action_type>
+                    </route_v2_customers_id_addresses_id>
                 </routes>
+                <attributes translate="ids" module="api2">
```
Comment (augsteyer) [inline src/app/code/community/Shopgate/Cloudapi/etc/api2.xml]: there is already an attributes node above (maybe did not notice in merge?), it needs to be moved up.

File: tests/postman/newman/collection.json [hunk 3]
```diff
@@ -13721,6 +13721,667 @@
...
+                    "method": "DELETE",
+                    "header": [
+                      {
+                        "key": "Content-Type",
+                        "value": "application/json"
+                      }
+                    ],
+                    "body": {
+                      "mode": "raw",
+                      "raw": "{\n\t\t\"prefix\": \"Test1.prefix\",\n        \"firstname\": \"Test1.first\",\n        \"middlename\": \"Test1.mid\",\n        \"lastname\": \"Test1.last\",\n        \"suffix\": \"Test1.suffix\",\n        \"company\": \"Test1.comapny\",\n        \"city\": \"Test1.city\",\n        \"country_id\": \"US\",\n        \"region\": \"Arizona\",\n        \"postcode\": \"85000\",\n        \"telephone\": \"618-322-0000\",\n        \"vat_id\": \"Test1.vat_id\",\n        \"street\": [\n            \"Test1.street1\",\n            \"Test1.street2\"\n        ]\n\t}"
```
Comment (augsteyer) [inline tests/postman/newman/collection.json]: why do your deletes have a body?

File: tests/postman/newman/collection.json [hunk 4]
```diff
@@ -13721,6 +13721,667 @@
...
+                }
+              ],
+              "_postman_isSubFolder": true
+            },
+            {
+              "name": "Multiple",
+              "description": "",
+              "item": [
+                {
+                  "name": "Get customer ID via ME Copy",
```
Comment (augsteyer) [inline tests/postman/newman/collection.json]: remove copy

File: tests/postman/newman/collection.json (hunk 3 above)
Comment (andre-kraus) [reply tests/postman/newman/collection.json]: Copy&paste error, removed.

**Pull request merged**